### ISSUE TITLE
BIOS updates Xiaomi Air

### ISSUE THREAD

**Issue by ghost** [Golovin-Andrey/xiaomi-mi-13-ubuntu#6]
**My question is: Did anybody try upgrading their Xiaomi INSYDE Corp. BIOS yet?**

Xiaomi after-sales is non-existent, especially in the "international" area. But to be fair, Xiaomi did make it very clear from the start that the Air in their opinion is a Chinese market only product. Outside China distributors (e.g. Gearbest, BangGood) love to sell you the Air, but play conveniently dumb on the support subject.

Trawling, I did find Air BIOS upgrades on the Xiaomi forums (Chinese and English), but haven't dared it yet, due to fears of bricking. I realise that I've a Skylake 12.5 (on X16.04LTS without any issues), while most here seem to sport a 13.3.
Nevertheless, it would be nice if some grey beards could give a thought, and help the *buntu/ linux Xiaomi users out there with info and feedback.

My 2 cents - what I found:
**For the Xiaomi Air 13 (i5-6200U) Skylake**
[TM 1613 - A06](https://mega.nz/#!bMRlCLST!5o7WVr-0cZx4mWJPE_cMCFGJtYyXJj9yDg6PBloR9sk)

**For the Xiaomi Air 12.5 (m3-6Y30) Skylake**
[TM1612 - A05](https://mega.nz/#!fY5lARYK!ytt5obuI4uUy1V3Vs1H-0WF-GWhUet7_ozGfbUxCgGc)

**For the Xiaomi 13.3 - Kabylake 2017 version**
[TM1604 - P0304](https://mega.nz/#!6MQlwZbT!nydgjZ2OnSxJ5W_AXK4KNTPbMoP1b6vLLdmDqQNykis)

[The original (Chinese, so Google Translate needed!) source](http://bbs.xiaomi.cn/t-13426797)
[The forum at TechTablets also has several Xiaomi users discussing and sharing BIOS](https://techtablets.com/forum/topic/bios-for-xiaomi-air-12-5-6y30/)

Since Xiaomi still needs DOS dependant flashing, [I assume you need some bootable USB with e.g. FreeDos](https://help.ubuntu.com/community/BIOSUpdate). Correct TM number can be found with `sudo dmidecode | less`:

`BIOS Information`
`        Vendor: INSYDE Corp.`
`        Version: A04`
`        Release Date: 08/06/2016`
[...]
`System Information`
`        Manufacturer: Timi`
 `       Product Name: TM1612`

_**For people landing here, looking for Xiaomi Air BIOS updates: please note that I don't know anything about the quality of these sources, nor can I say anything about their success! This is what we're discussing here! So IF you continue flashing your box, please do so AFTER ample contemplation of potential consequences!"**_

**Comment by Golovin-Andrey** [comment]
@BelladonnavGF, thanks for info. Do you know any particular reason to flash new bios? 
As for my self I would prefer "don't touch till it works"

**Comment by ghost** [comment]
@Golovin-Andrey, you're absolutely right of course. "If it's not broken, don't fix it" is a very wise and true paradigm. And I must say that my *buntu XAir gives me absolutely no troubles whatsoever. The contrary actually. But, having said that, there is always room for improvement. As you know, quite often power management is one of the first, most frequent things improved with BIOS/ UEFI upgrades. Others are memory management or additional hardware support, like for example that new, additional SSD that fills that empty slot...
But...
Yes, I can't argue with the solid wisdom of your remark: if all is well, the need for BIOS/ UEFI upgrades is questionable. And this seems supported by the fact that **nowhere** I found accounts on **what** the new Xiaomi BIOS version changed. 
That was why I was curious whether a grey beard, or even a clean shaven youngster, could share their experience after upgrade with us...

**Comment by eminfedar** [comment]
Hello,

I have 6200u version and I can't enter bios now. Is there a way to reset it? You shared some bios links. Is there anything like recovery or new bios with no password? (My password is corrupt, i can't login into bios now :))

**Comment by postadelmaga** [comment]
Any direct link to the official bios page ?

**Comment by migwell2** [comment]
After changing the battery of my XIAOMI AIR 12.5 from 2016 TM1612, I need to update the bios to recognize the new battery. My bios is in A02. I can't found the bios update A04 or A05, necessary to be able to use the new battery. Can you give me a download link or tell me who to ask.

**Comment by Golovin-Andrey** [comment]
I added couple BIOS binaries from 4PDA forum, didn't test it.

**Comment by migwell2** [comment]
I have download it from 4PDA and flash my bios. It works. The new battery is defective. Many thanks.

**Comment by JesseTsing** [comment]
https://github.com/johnnync13/Xiaomi-Notebook-Air-1Gen/tree/master/BIOS  this guy helps a lot!!!!!! Thanks to him!!!!! And he provides other notebook models..

**Comment by A3XES** [comment]
hi,can anybody reupload the bios file for the tm1604?
thnx

**Comment by bimbirim** [comment]
2 years ago I set a password on the BIOS. And already forgot.
Now I am looking for DUMP BIOS  on the model

Xiaomi Mi Notebook Air 12.5 7Y30
motherboard: timi tm1607

**Comment by tholyir** [comment]
TM

> hi,can anybody reupload the bios file for the tm1604? thnx

TM1604 Bios Version: XMAKB3M0P1911
https://drive.google.com/file/d/1-EjTL4t6afw91C9EsHsIfR5NX4ctt4bA/view?usp=sharing
